security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -736,53 +681,78 @@ func (sc *SecretCache) isTokenExpired() bool {
 	return false
 }
 
-func constructCSRHostName(trustDomain, token string) (string, error) {
-	// If token is jwt format, construct host name from jwt with format like spiffe://cluster.local/ns/foo/sa/sleep,
-	strs := strings.Split(token, ".")
-	if len(strs) != 3 {
-		return "", fmt.Errorf("invalid k8s jwt token")
-	}
+// sendRetriableRequest sends retriable requests for either CSR or ExchangeToken.
+// Prior to sending the request, it also sleep random millisecond to avoid thundering herd problem.
+func (sc *SecretCache) sendRetriableRequest(ctx context.Context, csrPEM []byte, providedExchangedToken, resourceName string, isCSR bool) ([]string, error) {
+	backOffInMilliSec := rand.Int63n(sc.configOptions.InitialBackoff)
+	cacheLog.Debugf("Wait for %d millisec for initial CSR", backOffInMilliSec)
```

Comment:
update log msg since now it's not only csr ?

Reply:
add comments below

Reply:
Updated. How does it look now? I don't see any comments below.